server/task/src/main/scala/org/apache/james/task/eventsourcing/EventSourcingTaskManager.scala

```diff
@@ -37,11 +37,10 @@ class EventSourcingTaskManager @Inject @VisibleForTesting private[eventsourcing]
   private val workQueue: WorkQueue = WorkQueue.builder().worker(new SerialTaskManagerWorker)
   private val delayBetweenPollingInMs = 500
 
-
   private def workDispatcher: Subscriber = {
     case Created(aggregateId, _, task) =>
       val taskWithId = new TaskWithId(aggregateId.taskId, task)
-      workQueue.submit(taskWithId, new ExecutionDetailsProjectionWorkerListener(executionDetailsProjection.updater, taskWithId.getId))
+      workQueue.submit(taskWithId, new WorkerStatusListener(taskWithId.getId, eventSourcingSystem))
```

Comment:
As far as I understand, you changed the design to generate comments from WorkQueue events.

Events represent things that happened. WorkQueue emits events because it describe what happened. Changing that into a command would mean: knowing that these events happened, please execute this command in this aggregate. What does it mean to request an aggregate to fail a Task? What if the aggregate implement a business rule that make that command fail? How would you handle it?

Reply:
So the event system should expose the event bus, and we would push the events directly into it without using commands.
I prefer the current approach which avoid exposing the bus, even if it means to handle commands which exists only to convey the message.

Reply:
False dychotomy. Before yesterday refactoring, we neither exposed the event bus nor generate non-business commands.

Reply:
Yes, but we created another false event bus. With events not persisted in the store.
The store contained only the Created and CancelRequest events, clearly not enough to reconstruct the projections by replaying the events.

We need to push all the events in the event bus of the event system. 
Either as we have done using commands or by pushing the other events directly in the eventbus.

Reply:
@mbaechler Why `Start`, `Fail`, `Complete` and `Cancel` are out of the business?